shared/src/main/scala/coop/rchain/pcasper/finalization/DagData.scala

```diff
@@ -0,0 +1,8 @@
+package coop.rchain.pcasper.finalization
+
+trait DagData[F[_], M, S] {
+  def witnessesF: M => F[Map[S, M]]
+  def justificationsF: M => F[Map[S, M]]
```

Comment:
Just a question. What are `witnesses` and `justifications`? Justifications are used frequently in the code, so it would be good to get a clear understanding of what it is. Is it what the new message is based on?

Reply:
Justifications are latest messages from all bonded validators seen by the message. The arrows you see on DAGs expose justifications.
<img width="378" alt="Screenshot 2022-02-21 at 14 50 11" src="https://user-images.githubusercontent.com/1746367/154949912-8d18afd9-8eb7-4867-a5a2-e026a583f39e.png">
Witness from sender is the first message produced by a sender that has target in the view.
E.g. here red W is a witness for all green S's.
<img width="98" alt="Screenshot 2022-02-21 at 14 52 04" src="https://user-images.githubusercontent.com/1746367/154950175-85ebfa6a-49a8-422d-8835-617e3dc0ff57.png">